rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -222,30 +222,33 @@ public Wallet getNoSpendWalletForLiveFederations() throws IOException {
     /**
      * In case of a lock tx: Transfers some SBTCs to the sender of the btc tx and keeps track of the new UTXOs available for spending.
      * In case of a release tx: Keeps track of the change UTXOs, now available for spending.
-     * @param btcTx The bitcoin transaction
-     * @param height The height of the bitcoin block that contains the tx
-     * @param pmt Partial Merklee Tree that proves the tx is included in the btc block
+     * @param rskTx The RSK transaction
+     * @param btcTxSerialized The raw BTC tx
+     * @param height The height of the BTC block that contains the tx
+     * @param pmtSerialized The raw partial Merkle tree
      * @throws BlockStoreException
      * @throws IOException
      */
-    public void registerBtcTransaction(Transaction rskTx, BtcTransaction btcTx, int height, PartialMerkleTree pmt) throws BlockStoreException, IOException {
+    public void registerBtcTransaction(Transaction rskTx, byte[] btcTxSerialized, int height, byte[] pmtSerialized) throws IOException, BlockStoreException {
         Context.propagate(btcContext);
-
-        Federation federation = getActiveFederation();
-
+        Sha256Hash btcTxHash = BridgeUtils.calculateBtcTxHash(btcTxSerialized);
         // Check the tx was not already processed
-        if (provider.getBtcTxHashesAlreadyProcessed().keySet().contains(btcTx.getHash())) {
+        if (provider.getBtcTxHashesAlreadyProcessed().keySet().contains(btcTxHash)) {
             logger.warn("Supplied tx was already processed");
             return;
         }
 
-        // Check the tx is in the partial merkle tree
-        List<Sha256Hash> hashesInPmt = new ArrayList<>();
-        Sha256Hash merkleRoot = pmt.getTxnHashAndMerkleRoot(hashesInPmt);
-        if (!hashesInPmt.contains(btcTx.getHash())) {
-            logger.warn("Supplied tx is not in the supplied partial merkle tree");
-            panicProcessor.panic("btclock", "Supplied tx is not in the supplied partial merkle tree");
-            return;
+        Sha256Hash merkleRoot;
+        try {
+            PartialMerkleTree pmt = new PartialMerkleTree(bridgeConstants.getBtcParams(), pmtSerialized, 0);
+            List<Sha256Hash> hashesInPmt = new ArrayList<>();
+            merkleRoot = pmt.getTxnHashAndMerkleRoot(hashesInPmt);
+            if (!hashesInPmt.contains(btcTxHash)) {
+                logger.warn("Supplied tx is not in the supplied partial merkle tree");
+                return;
```

Comment:
throw?

Reply:
I think throwing an exception here will result in a _hard fork_ b/c the transaction will end up aborted